### PR TITLE
Add projected fields of companies in the auth definition file

### PR DESCRIPTION
**Pull request by LucasBrazi06**

### CODE REVIEW

File: src/types/UserToken.ts [hunk 1]
```diff
@@ -3,6 +3,7 @@ import User from './User';
 export default interface UserToken {
   id?: string;
   role?: string;
+  rolesACL?: string[];
```
Comment (jerome-benoit) [inline src/types/UserToken.ts:6]: It does not seem to be used.

File: src/server/rest/v1/service/SiteService.ts [hunk 2]
```diff
@@ -600,7 +601,7 @@ export default class SiteService {
     // Check data is valid
     UtilsService.checkIfSiteValid(filteredRequest, req);
     // Check static auth for reading company
-    if (!Authorizations.canReadCompany(req.user)) {
+    if (!(await Authorizations.canReadCompany(req.user)).authorized) {
```
Comment (jerome-benoit) [inline src/server/rest/v1/service/SiteService.ts:604]: Is `.authorized` really needed here?

File: src/types/UserToken.ts (hunk 1 above)
Comment (LucasBrazi06) [reply src/types/UserToken.ts:6]: Yes it is used in check of auths right away without rebuilding it on the fly whenever there is an HTTP request:

image

File: src/server/rest/v1/service/SiteService.ts (hunk 2 above)
Comment (LucasBrazi06) [reply src/server/rest/v1/service/SiteService.ts]: This code is no longer here after merge with master-qa :-)

**Comment by LucasBrazi06** [comment]
@roxiciobotea210 I merged the pull request to keep the next changes readable and I think the current implementation is stable enough.

**Pull request merged**